### DISCUSSION TITLE
Responsive Light/Dark Mode

### DISCUSSION THREAD

**Discussion by randose** [microsoft/fluentui#28410] [General] [answered]
Build an Excel Add-In and as recommended, I'm using FluentUI v9. I'd like my theme to automatically switch between a dark theme and light theme according to the user's setting in Excel, or at least according to their OS setting. I'd also like to give the user the choice of manually selecting dark or light mode in my taskpane app's settings panel. 

Does FluentUI support this?

**Comment by behowell** [comment, marked as answer]
Hi @randose, no FluentUI v9 doesn't provide this behavior. You'll need to add your own logic to pick between the dark or light theme that you pass to the FluentProvider. Check the Excel add-in documentation to see if there's a way to detect dark/light mode. Or, here's a stackoverflow answer about how to detect the user's OS theme in JavaScript: https://stackoverflow.com/questions/56393880/how-do-i-detect-dark-mode-using-javascript#answers
